aws/resource_aws_lex_slot_type.go

```diff
@@ -0,0 +1,249 @@
+							MinItems: 1,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validation.StringLenBetween(1, 140),
+							},
+						},
+						"value": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(1, 140),
+						},
+					},
+				},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 100),
+					validation.StringMatch(regexp.MustCompile(`^((AMAZON\.)_?|[A-Za-z]_?)+`), ""),
+				),
+			},
+			"value_selection_strategy": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  lexmodelbuildingservice.SlotValueSelectionStrategyOriginalValue,
+				ValidateFunc: validation.StringInSlice([]string{
+					lexmodelbuildingservice.SlotValueSelectionStrategyOriginalValue,
+					lexmodelbuildingservice.SlotValueSelectionStrategyTopResolution,
+				}, false),
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsLexSlotTypeCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lexmodelconn
+	name := d.Get("name").(string)
+
+	input := &lexmodelbuildingservice.PutSlotTypeInput{
+		CreateVersion:          aws.Bool(d.Get("create_version").(bool)),
+		Description:            aws.String(d.Get("description").(string)),
+		Name:                   aws.String(name),
+		ValueSelectionStrategy: aws.String(d.Get("value_selection_strategy").(string)),
+	}
+
+	if v, ok := d.GetOk("enumeration_value"); ok {
+		input.EnumerationValues = expandLexEnumerationValues(v.(*schema.Set).List())
+	}
+
+	err := resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		output, err := conn.PutSlotType(input)
+
+		if isAWSErr(err, lexmodelbuildingservice.ErrCodeConflictException, "") {
+			input.Checksum = output.Checksum
+			return resource.RetryableError(fmt.Errorf("%q: slot type still creating", d.Id()))
+		}
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("error creating slot type %s: %s", name, err)
+	}
+
+	d.SetId(name)
+
+	return resourceAwsLexSlotTypeRead(d, meta)
+}
+
+func resourceAwsLexSlotTypeRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lexmodelconn
+
+	resp, err := conn.GetSlotType(&lexmodelbuildingservice.GetSlotTypeInput{
+		Name:    aws.String(d.Id()),
+		Version: aws.String("$LATEST"),
+	})
+	if isAWSErr(err, lexmodelbuildingservice.ErrCodeNotFoundException, "") {
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("error getting slot type %s: %s", d.Id(), err)
+	}
+
+	d.Set("checksum", resp.Checksum)
+	d.Set("description", resp.Description)
+	d.Set("name", resp.Name)
+	d.Set("value_selection_strategy", resp.ValueSelectionStrategy)
+	d.Set("version", "$LATEST")
```

Comment:
So, I believe the problem is that if we store the actual version in state and the user has $LATEST in their config, it will always show a diff on apply. How should we handle this?

Reply:
Since the `version` value is read-only, this shouldn't be a problem